xtokens/src/tests.rs

```diff
@@ -1048,3 +1048,54 @@ fn specifying_a_non_existent_asset_index_should_fail() {
 		);
 	});
 }
+
+#[test]
+fn send_with_zero_amount() {
+	TestNet::reset();
+
+	ParaA::execute_with(|| {
+		assert_noop!(
+			ParaXTokens::transfer(
+				Some(ALICE).into(),
+				CurrencyId::B,
+				0,
+				Box::new(
+					(
+						Parent,
+						Parachain(2),
+						Junction::AccountId32 {
+							network: NetworkId::Any,
+							id: BOB.into(),
+						},
+					)
+						.into()
+				),
+				40,
+			),
+			Error::<para::Runtime>::ZeroAmount
+		);
+
+		assert_noop!(
+			ParaXTokens::transfer_multicurrencies(
+				Some(ALICE).into(),
+				vec![(CurrencyId::B, 0), (CurrencyId::B1, 50)],
+				1,
+				Box::new(
+					(
+						Parent,
+						Parachain(2),
+						Junction::AccountId32 {
+							network: NetworkId::Any,
+							id: BOB.into(),
+						},
+					)
+						.into()
+				),
+				40,
+			),
+			Error::<para::Runtime>::ZeroAmount
+		);
+	});
+
+	// TODO: should have more tests after https://github.com/paritytech/polkadot/issues/4996
```

Comment:
should have more test to ensure MultiAsset is valid (fungible & non zero amount) after https://github.com/paritytech/polkadot/issues/4996